tests/unit/lib/package/test_artifact_exporter.py

```diff
@@ -507,6 +507,39 @@ class MockResourceNoForceZip(Resource):
         self.code_signer_mock.sign_package.assert_not_called()
         self.assertEqual(resource_dict[resource.PROPERTY_NAME], s3_url)
 
+    @patch("shutil.rmtree")
+    @patch("zipfile.is_zipfile")
+    @patch("samcli.lib.package.artifact_exporter.copy_to_temp_dir")
+    @patch("samcli.lib.package.artifact_exporter.zip_and_upload")
+    @patch("samcli.lib.package.artifact_exporter.is_local_file")
+    def test_resource_using_array_property_without_force_zip(
+        self, is_local_file_mock, zip_and_upload_mock, copy_to_temp_dir_mock, is_zipfile_mock, rmtree_mock
+    ):
+        class MockResourceNoForceZip(Resource):
+            PROPERTY_NAME = "foo.bar"
+
+        resource = MockResourceNoForceZip(self.s3_uploader_mock, self.code_signer_mock)
+
+        resource_id = "id"
+        original_path = "/path/to/file"
+        resource_dict = {"foo": [{"bar": original_path}]}
```

Comment:
You could maybe add more elements in this array:
- One with only a `baz` attribute
- One with `bar` and `baz`

This would help testing that we are going through all elements and not updating the baz attribute.

Reply:
Ok this is done in unit tests, resolving this